File: templates/java-webflux/subsystem/post/component/src/main/java/{{PACKAGE_NAME}}/post/PostController.java
package {{PACKAGE_NAME}}.post;

import {{PACKAGE_NAME}}.post.model.Post;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/posts")
class PostController {

  private PostService postService;

  PostController(PostService postService) {
    this.postService = postService;
  }

  @GetMapping
  Flux<Post> findAllPost() {
    return this.postService.findAll();
  }

  @GetMapping("/{id}")
  Mono<Post> findPostById(@PathVariable Integer id) {
    return this.postService.findById(id);
  }

}

File: templates/kotlin-webflux/subsystem/post/component/src/main/kotlin/{{PACKAGE_NAME}}/post/PostController.kt
package {{PACKAGE_NAME}}.post

import {{PACKAGE_NAME}}.post.model.Post
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

@RestController
@RequestMapping("/posts")
internal class PostController(
  private val postService: PostService
) {

  @GetMapping
  fun findAllPost(): Flux<Post> =
    this.postService.findAll()

  @GetMapping("/{id}")
  fun findPostById(@PathVariable id: Int): Mono<Post> =
    this.postService.findById(id)

}
